src/dls_bluesky_core/core/types.py

```diff
@@ -1,8 +1,11 @@
-from typing import Any, Callable, Generator
+from typing import Any, Callable, Generator, Union
 
 from bluesky import Msg
+from ophyd_async.core import Device
+from scanspec.specs import DURATION
 
 #  'A true "plan", usually the output of a generator function'
 MsgGenerator = Generator[Msg, Any, None]
 #  'A function that generates a plan'
 PlanGenerator = Callable[..., MsgGenerator]
+ScanAxis = Union[Device, DURATION]
```

Comment:
I think we do allow strings in some cases

```suggestion
ScanAxis = Union[str, Device, DURATION]
```

Reply:
Not in this case, we require it to have been filled in to a Device